pyfarm/agent/sysinfo/network.py

```diff
@@ -66,18 +66,20 @@ def mac_addresses(long_addresses=False, as_integers=False):
         When ``True`` convert all mac addresses to integers.
     """
     results = set()
-    for ifaces in map(netifaces.ifaddresses, netifaces.interfaces()):
-        for entry in ifaces.get(netifaces.AF_LINK, []):
-            mac = entry.get("addr", "")
-
-            if all([mac, not long_addresses, len(mac) == 17]) \
-               or all([long_addresses, mac, len(mac) >= 17]):
-                mac_as_int = int("0x" + mac.replace(":", ""), 0)
-                if mac_as_int != 0:
-                    if as_integers:
-                        results.add(mac_as_int)
-                    else:
-                        results.add(mac)
+    net_if_addrs = psutil.net_if_addrs()
+
+    for name, nics in net_if_addrs.items():
+        for nic in nics:
+            if nic.family != psutil.AF_LINK:
+                continue
+
+            if not long_addresses and len(nic.address) > 17:
+                continue
+
+            mac = nic.address
+            if as_integers:
+                mac = int("0x" + nic.address.replace(":", ""), 0)
+            results.add(mac)
```

Comment:
This code fails to filter out 00:00:00:00:00:00 mac addresses (from the loopback interface), as the old code did.